### PR TITLE
Articles can we viewed without SignUp

### LINKED ISSUE #319: bug : All users should be able to read the articles
**Describe the bug**
If we are not logged in, on clicking on articles in navbar, it prompts us to signIn or login.

**To Reproduce**

1. Click on Articles in the navbar
2. See prompt

**Expected behavior**
The logged out users must be able to read the articles.

**Screenshots**
None

**Additional context**
If clicked on create article, then the prompt must appear.

### PR DESCRIPTION
**Pull request by devansh-dek**
**What kind of change does this PR introduce?**
Now users can access all articles without being logged in.

**Issue Number:**  319

Fixes #319 
**Snapshots/Videos:**

https://github.com/Tanay-ErrorCode/lupo-skill/assets/83805376/5a8a5d9d-857c-4693-8e68-12e9f4e35bd7

Also added a toast telling users to login to view an article or to create an article

### CODE REVIEW

**Comment by Doraemon012** [comment]
@devansh-dek Good work but some changes are required.

1. The SignUp modal should not be visible when we open the articles page.
2. If the user clicks on create article button, and is not logged in, the sign up modal should be shown.
3. The user should also be able to read articles without being signed in.
4. There is no need to show toasts.

File: src/components/Article/ArticlesHomepage.tsx [hunk 1]
```diff
@@ -36,23 +37,42 @@ interface Article {
...
-    if (localStorage.getItem("userUid") == null) {
-      window.location.href = "#/";
-    }
     const fetchArticles = async () => {
       const articlesRef = ref(database, "articles");
       const userUid = localStorage.getItem("userUid");
 
       if (!userUid) {
+        try {
+          const snapshot = await get(articlesRef);
```
Comment (Doraemon012) [inline src/components/Article/ArticlesHomepage.tsx:51]: Why have you added this? Please explain.

File: src/components/Article/ArticlesHomepage.tsx [hunk 2]
```diff
@@ -87,6 +107,21 @@ const ArticlesHomepage: React.FC = () => {
     fetchArticles();
   }, []);
 
+  const handleArticleLinkClick = (e: any) => {
```
Comment (Doraemon012) [inline src/components/Article/ArticlesHomepage.tsx:110]: Remove this.

**Review by Doraemon012 — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: src/components/Article/ArticlesHomepage.tsx (hunk 2 above)
Comment (devansh-dek) [reply src/components/Article/ArticlesHomepage.tsx:110]: I added this so if an user not signed in trys to open any article he gets a message that he must sign up before viewing entire article. Do you want him to see article anyways?

File: src/components/Article/ArticlesHomepage.tsx (hunk 1 above)
Comment (devansh-dek) [reply src/components/Article/ArticlesHomepage.tsx:51]: I added it so we can fetch articles to show if user is not signed in

**Comment by devansh-dek** [comment]
Okay ill  do the necessary changes. Also what to do if someone clicks on article do i have to show him pop too?

**Comment by Tanay-ErrorCode** [comment]
> Okay ill do the necessary changes. Also what to do if someone clicks on article do i have to show him pop too?

@devansh-dek No, the article must be openable and readable without signin.

**Comment by devansh-dek** [comment]
Here is how it works now

Uploading Lupo Skill - Google Chrome 2024-07-08 13-23-25.mp4…

**Comment by Doraemon012** [comment]
> Here is how it works now
> 
> Uploading Lupo Skill - Google Chrome 2024-07-08 13-23-25.mp4…

@devansh-dek 
Please upload the screen recording.

**Comment by devansh-dek** [comment]
here is the recording

https://github.com/Tanay-ErrorCode/lupo-skill/assets/83805376/411a8d17-8f25-452f-bf97-0f44a7374d91

**Review by Doraemon012 — APPROVED** [review]

**Pull request merged**